### PR TITLE
Don't allow changing matrels directly

### PR DESCRIPTION
**Pull request by usmanm**

### CODE REVIEW

File: src/backend/utils/misc/pipelinedb.conf.sample [hunk 1]
```diff
@@ -627,6 +627,16 @@
 # each database
 #continuous_query_num_workers = 2
 
+# allow direct changes to be made to materialization tables?
+#continuous_query_materialization_table_updatable = off
```
Comment (derekjn) [inline src/backend/utils/misc/pipelinedb.conf.sample]: Let's leave it undocumented for now.

File: src/backend/executor/execMain.c [hunk 2]
```diff
@@ -968,6 +970,20 @@ CheckValidResultRel(Relation resultRel, CmdType operation)
 	switch (resultRel->rd_rel->relkind)
 	{
 		case RELKIND_RELATION:
+			if (!MatRelUpdatesEnabled())
```
Comment (derekjn) [inline src/backend/executor/execMain.c]: This is pedantic, but we might as well make this a macro.

**Comment by derekjn** [comment]
Looks good

File: src/backend/utils/misc/pipelinedb.conf.sample (hunk 1 above)
Comment (usmanm) [inline src/backend/utils/misc/pipelinedb.conf.sample]: Any reason to do so? Only power users ever really look at configuration options and it's prolly ok to expose this to them?

Comment (derekjn) [inline src/backend/utils/misc/pipelinedb.conf.sample]: Nope, feel free to leave it documented if you'd prefer.

**Pull request merged**